QuickFIXn/AbstractInitiator.cs

```diff
@@ -20,22 +22,43 @@ public abstract class AbstractInitiator : IInitiator
         private readonly SessionFactory _sessionFactory;
         private Thread? _thread;
 
-        protected readonly NonSessionLog _nonSessionLog;
+        protected readonly ILogger _nonSessionLog;
 
         public bool IsStopped { get; private set; } = true;
 
+        [Obsolete("Use \"Microsoft.Extensions.Logging.ILoggerFactory\" instead of \"QuickFix.Logger.ILogFactory\".")]
         protected AbstractInitiator(
             IApplication app,
             IMessageStoreFactory storeFactory,
             SessionSettings settings,
-            ILogFactory? logFactoryNullable,
-            IMessageFactory? messageFactoryNullable)
+            ILogFactory? logFactoryNullable = null,
+            IMessageFactory? messageFactoryNullable = null)
         {
             _settings = settings;
-            var logFactory = logFactoryNullable ?? new NullLogFactory();
+            ILoggerFactory logFactory = logFactoryNullable is null
+                ? NullLoggerFactory.Instance
+                : new LogFactoryAdapter(logFactoryNullable, settings);
             var msgFactory = messageFactoryNullable ?? new DefaultMessageFactory();
             _sessionFactory = new SessionFactory(app, storeFactory, logFactory, msgFactory);
-            _nonSessionLog = new NonSessionLog(logFactory);
+            _nonSessionLog = logFactory.CreateLogger("QuickFix");
+
+            HashSet<SessionID> definedSessions = _settings.GetSessions();
+            if (0 == definedSessions.Count)
+                throw new ConfigError("No sessions defined");
+        }
+
+        protected AbstractInitiator(
+            IApplication app,
+            IMessageStoreFactory storeFactory,
+            SessionSettings settings,
+            ILoggerFactory? logFactoryNullable = null,
+            IMessageFactory? messageFactoryNullable = null)
+        {
+            _settings = settings;
+            var logFactory = logFactoryNullable ?? new NullLoggerFactory();
```

Comment:
it won't let me submit a suggestion through the UI, so here it is the old school way:

```diff
diff --git a/QuickFIXn/AbstractInitiator.cs b/QuickFIXn/AbstractInitiator.cs
index 4a3a9700..d6528b9c 100644
--- a/QuickFIXn/AbstractInitiator.cs
+++ b/QuickFIXn/AbstractInitiator.cs
@@ -31,31 +31,28 @@ namespace QuickFix
             IApplication app,
             IMessageStoreFactory storeFactory,
             SessionSettings settings,
-            ILogFactory? logFactoryNullable = null,
-            IMessageFactory? messageFactoryNullable = null)
+            ILogFactory? logFactoryNullable,
+            IMessageFactory? messageFactoryNullable)
+            : this(
+            app,
+            storeFactory,
+            settings,
+            logFactoryNullable is null
+                  ? NullLoggerFactory.Instance
+                  : new LogFactoryAdapter(logFactoryNullable, settings),
+            messageFactoryNullable)
         {
-            _settings = settings;
-            ILoggerFactory logFactory = logFactoryNullable is null
-                ? NullLoggerFactory.Instance
-                : new LogFactoryAdapter(logFactoryNullable, settings);
-            var msgFactory = messageFactoryNullable ?? new DefaultMessageFactory();
-            _sessionFactory = new SessionFactory(app, storeFactory, logFactory, msgFactory);
-            _nonSessionLog = logFactory.CreateLogger("QuickFix");
-
-            HashSet<SessionID> definedSessions = _settings.GetSessions();
-            if (0 == definedSessions.Count)
-                throw new ConfigError("No sessions defined");
         }

         protected AbstractInitiator(
             IApplication app,
             IMessageStoreFactory storeFactory,
             SessionSettings settings,
-            ILoggerFactory? logFactoryNullable = null,
-            IMessageFactory? messageFactoryNullable = null)
+            ILoggerFactory? logFactoryNullable,
+            IMessageFactory? messageFactoryNullable)
         {
             _settings = settings;
-            var logFactory = logFactoryNullable ?? new NullLoggerFactory();
+            var logFactory = logFactoryNullable ?? NullLoggerFactory.Instance;
             var msgFactory = messageFactoryNullable ?? new DefaultMessageFactory();
             _sessionFactory = new SessionFactory(app, storeFactory, logFactory, msgFactory);
             _nonSessionLog = logFactory.CreateLogger("QuickFix");
```